core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -429,6 +429,25 @@ public MessageBus getMessageBus() {
         return messageBus;
     }
 
+    public TableMetadata getLegacyMetadata(TableToken tableToken) {
+        return getLegacyMetadata(tableToken, TableUtils.ANY_TABLE_VERSION);
+    }
+
+        /**
+         * Retrieves up-to-date table metadata regardless of table type.
+         *
+         * @param tableToken table token
+         * @param desiredVersion version of table metadata used previously if consistent metadata reads are required
+         * @return returns {@link io.questdb.cairo.wal.seq.SequencerMetadata} for WAL tables and {@link TableMetadata}
+         * for non-WAL, which would be metadata of the {@link TableReader}
+         */
```

Comment:
nit: formatting indentation is off

Reply:
formatted in https://github.com/questdb/questdb/pull/4044/commits/1b26f6db43a998a89649ecb8f103a9fc2740ad43